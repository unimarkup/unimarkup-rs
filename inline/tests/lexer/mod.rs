use crate::Snapshot;
use unimarkup_commons::test_runner::{self, snap_test_runner::SnapTestRunner};
use unimarkup_inline::Tokenize;

mod snapshot;

#[test]
fn test_lexer_snapshots() {
    for case in crate::prepare_test_cases("spec/markup", "spec/snapshots/lexer") {
        let symbols = test_runner::scan_str(&case.input);
        let runner = SnapTestRunner::with_fn(&case.name, &symbols, |symbols| {
            let rest = &[];
            let snapshot = Snapshot::snap((case.input.as_ref(), symbols.tokens()));
            (snapshot, rest)
        })
        .with_info(format!("Test '{}' from '{}'", case.name, case.file_name));

        unimarkup_commons::run_snap_test!(runner, &case.out_path);
    }
}
