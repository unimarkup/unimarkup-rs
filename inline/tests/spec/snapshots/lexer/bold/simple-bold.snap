---
source: inline/tests/lexer/mod.rs
info: "Test 'simple-bold' from: 'bold.yml'"
---
**Bold**
**
^^ -> Bold @ (1:1)->(1:3)
  Bold
  ^^^^ -> Plain @ (1:3)->(1:7)
      **
      ^^ -> Bold @ (1:7)->(1:9)
        ⏎
        ^ -> Newline @ (1:9)->(2:1)

---
With input:

**Bold**


